src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -499,7 +501,7 @@ void DeleteInvalidBlocks(Keccak invalidBlockHash)
             List<Block> blocksToProcess = processingBranch.BlocksToProcess;
             if (options.ContainsFlag(ProcessingOptions.ForceProcessing))
             {
-                processingBranch.Blocks.Clear();
+                processingBranch.Blocks.Clear(); // TODO: investigate why if we clear it all we need to collect and iterate on all the blocks in PrepareProcessingBranch?
```

Comment:
Yes this could be moved to not create the branch in `PrepareProcessingBranch`, can be done in different PR.